### PR TITLE
Allow null flutter schema in pubspec

### LINKED ISSUE #14008: Error detected in pubspec.yaml: #/properties/flutter: type: wanted [object] got null
## Steps to Reproduce

When I build my project, I get this error, the project still builds fine and I'm able to develop but the error shows up in the logs.

```
Launching lib/main.dart on iPhone X in debug mode...
Error detected in pubspec.yaml:
#/properties/flutter: type: wanted [object] got null

```

This is my pubspec

```
name: venflutter
description: Ven Chat App

dependencies:
  shared_preferences: "^0.3.0"
  jaguar_client: "^0.2.1"
  jaguar_serializer: "^0.5.1"
  event_bus: any
  flutter_secure_storage: "^0.0.1"
  fluro: "^1.2.1"
  cached_network_image: "^0.2.1"
  google_sign_in: 0.3.1
  firebase_analytics: 0.0.4
  firebase_auth: 0.2.0
  firebase_database: 0.0.12
  firebase_storage: 0.0.5
  flutter_image: "^0.0.2"
  async_loader: "^0.1.0"
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.0

dev_dependencies:
  flutter_test:
    sdk: flutter


# For information on the generic Dart part of this file, see the
# following page: https://www.dartlang.org/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  assets:
    - assets/splash.jpg

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.io/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.io/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies, 
  # see https://flutter.io/custom-fonts/#from-packages
```

## Logs

```
Launching lib/main.dart on iPhone X in debug mode...
Error detected in pubspec.yaml:
#/properties/flutter: type: wanted [object] got null
Running Xcode build...
[Firebase/Core][I-COR000003] The default Firebase app has not yet been configured. Add [FIRApp configure] to your application initialization. Read more: https://goo.gl/ctyzm8.
[Firebase/Analytics][I-ACS005000] The AdSupport Framework is not currently linked. Some features will not function properly. Learn more at http://goo.gl/9vSsPb
[Firebase/Analytics][I-ACS023007] Firebase Analytics v.3900000 started
[Firebase/Analytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
[Firebase/Messaging][I-IID001001] FIRInstanceID AppDelegate proxy enabled, will swizzle app delegate remote notification handlers. To disable add "FirebaseAppDelegateProxyEnabled" to your Info.plist and set it to NO
[Firebase/Analytics][I-ACS003007] Successfully created Firebase Analytics App Delegate Proxy automatically. To disable the proxy, set the flag FirebaseAppDelegateProxyEnabled to NO in the Info.plist
#/properties/flutter: type: wanted [object] got null
Error detected in pubspec.yaml:
[Firebase/Analytics][I-ACS032003] iAd framework is not linked. Search Ad Attribution Reporter is disabled.
[Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
Syncing files to device iPhone X...

```

## Flutter Doctor

flutter doctor
[✓] Flutter (on Mac OS X 10.13.2 17C88, locale en-US, channel master)
    • Flutter at /Users/faisalabid/Development/flutter
    • Framework revision 3782415d15 (4 days ago), 2018-01-05 14:48:50 -0800
    • Engine revision 6a724f0d3e
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.5b56dfb77a76e9d8021086c0684b1e8bd9dbca9a

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /Users/faisalabid/Library/Android/sdk
    • Android NDK at /Users/faisalabid/Library/Android/sdk/ndk-bundle
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.2, Build version 9C40b
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 3.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b08)

[✓] IntelliJ IDEA Ultimate Edition (version 2017.2.6)
    • Flutter plugin version 19.1
    • Dart plugin version 172.4343.25

[✓] Connected devices
    • Pixel XL • HT6A60208436                         • android-arm • Android 8.1.0 (API 27)
    • iPhone X • 25D3677C-3836-4267-9A41-D68F3666E8B6 • ios         • iOS 11.2 (simulator)

### PR DESCRIPTION
**Pull request by xqwzts**
Fixes #14008 by allowing  the `flutter:` section of the pubspec schema to be null.

### CODE REVIEW

**Comment by xqwzts** [comment]
[please don't merge, will add a test in a bit]

**Comment by xqwzts** [comment]
Added a test case.

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

(Will merge when the tree is green)

**Comment by xqwzts** [comment]
Thanks!

**Pull request merged**
